4. Stack/destroy_asteroid/DestroyAsteroidArnab.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * DestroyAsteroidArnab.java
+ * Given an array of asteroid sizes with positive and negative signs,positive meaning travelling towards right and negative
+ * means trallening towards left.Asked to find out the final state of the asteroids as to which asteroids will remain 
+ * after collision if asteroids travel with same speed and after collision smaller size asteroid gets abloished.
+ * Description:-
+ * Create a stack which will store asteroids travelling in a particular direction,as soon as a asteroid travelling in a 
+ * different direction is encountered, compare the peek value of stack with that as to if that asteroid is interferring with the 
+ * previous asteroid.If yes,then check which will destroy and which will remain,update the stack accordingly and then proceed
+ * After the loop the asteroid present in the stack are the ones to remain after collision.Use a ArrayList to print the 
+ * Stack in reverse order.
+ * Time Complexity-O(n) Extra Space Complexity-O(n)
+ * @author [codebook-2000] (https://github.com/codebook-2000)
+ */
+ 
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Stack;
+
+public class DestroyAsteroidArnab {
+    static ArrayList<Integer> solve(int[] arr, int n) {
+        Stack<Integer> st = new Stack<Integer>(); //Creating the stack
+        int j = 0;
+
+        while (j < n)       //Starting the loop
+        {
+            if (st.isEmpty() == true) {  //If stack is empty,then whatever negative or positive push to stack
```

Comment:
If we omit it,then it might give error when it would calculate for st.peek() because the stack has got empty!
To deal with that situation, that extra Empty() condition is checked.

Reply:
I mean change it to if (st.isEmpty())